presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -7859,6 +7863,26 @@ public void testQuantifiedComparison()
         assertQuery("SELECT cast(1 as decimal(3,2)) <> ANY(SELECT cast(1 as decimal(3,1)))");
     }
 
+    @Test(dataProvider = "quantified_comparisons_corner_cases")
+    public void testQuantifiedComparisonCornerCases(String query) {
+        assertQuery(query);
+    }
+
+    @DataProvider(name = "quantified_comparisons_corner_cases")
+    public Object[][] qualifiedComparisonsCornerCases() {
+        List<QueryTemplate.Parameter> subquery = new QueryTemplate.Parameter("subquery").of(
+                "SELECT * FROM (SELECT 1 WHERE false) as empty_table",
+                "SELECT CAST(NULL AS INTEGER) AS null_singleton",
+                "SELECT * FROM (VALUES (1), (NULL)) AS mixed_values"
+        );
+        List<QueryTemplate.Parameter> quantifier = new QueryTemplate.Parameter("quantifier").of("ALL", "ANY");
+        List<QueryTemplate.Parameter> lhs = new QueryTemplate.Parameter("lhs").of("1", "NULL");
+        List<QueryTemplate.Parameter> relation = new QueryTemplate.Parameter("relation").of("=", "!=", "<", ">", "<=", ">=");
+        QueryTemplate query = new QueryTemplate("SELECT %lhs% %relation% %quantifier% (%subquery%)");
+        Stream<Object[]> queries = query.replaceAll(subquery, quantifier, lhs, relation).map((String q) -> new Object[] { q });
```

Comment:
`Arguments` is just a util class, returning Lists and Object[][] - not Arguments instances. Maybe a DSL for creating test arguments could be written - but I'd say let's wait with that till we migrate to junit5 ;)